src/unaudited/EcdsaEpochRegistry_Permissionless.sol

```diff
@@ -0,0 +1,250 @@
+        address[] memory /*operatorsToEvaluate*/,
+        uint256 /*minimumWeightRequirement*/
+    ) public virtual override {
+        uint256 _currentEpoch = currentEpoch();
+        require(operatorsForEpoch[_currentEpoch].length == 0, "operator set already calculated for current epoch");
+
+        uint256 _numberRegisteredOperators = numberRegisteredOperators();
+        uint256 _realizedMinimumWeight = type(uint256).max;
+
+        for (uint256 i = 0; i < _numberRegisteredOperators; ++i) {
+            address operator = registeredOperators[i];
+            uint256 operatorWeight = weightOfOperator(operator);
+            // add any registered operator that meets the weight requirement to the current epoch's operator set
+            if (operatorWeight >= minimumWeightRequirement) {
+                operatorsForEpoch[_currentEpoch].push(operator);
+                _operatorStakeHistory[operator][_currentEpoch] = operatorWeight;
+                // TODO: add event
+                if (operatorWeight < _realizedMinimumWeight) {
+                    _realizedMinimumWeight = operatorWeight;
+                }
+            // otherwise, evict the operator
+            } else {
+                _deregisterOperator(operator);
+            }
+        }
+
+        // require(operatorsForEpoch[_currentEpoch].length != 0, "no operators found for current epoch");
+        realizedMinimumWeight[_currentEpoch] = _realizedMinimumWeight;
+
+        // perform exponential retargeting -- adjust `minimumWeightRequirement` to target the desired number of operators
+        if (_currentEpoch != 0) {
+            _adjustMinimumWeight({previousEpoch: _currentEpoch - 1});
+        }
+    }
+
+    /*******************************************************************************
+                            INTERNAL FUNCTIONS
+    *******************************************************************************/
+    // TODO: documentation
+    function _registerOperator(
+        address operator,
+        ISignatureUtils.SignatureWithSaltAndExpiry memory operatorSignature
+    ) internal virtual override {
+        if (operatorStatus[operator] != OperatorStatus.REGISTERED) {
+            require(weightOfOperator(operator) >= minimumWeightRequirement,
+                "insufficient weight to register");
+            operatorStatus[operator] = OperatorStatus.REGISTERED;
+
+            // Register the operator with the EigenLayer via this AVS's ServiceManager
+            serviceManager.registerOperatorToAVS(operator, operatorSignature);
+
+            // TODO: event
+            // emit OperatorRegistered(operator, operatorId);
+
+            operatorIndex[operator] = numberRegisteredOperators();
+            registeredOperators.push(operator);
+        }
+    }
+
+    // TODO: documentation
+    function _deregisterOperator(
+        address operator
+    ) internal virtual override {
+        if (operatorStatus[operator] == OperatorStatus.REGISTERED) {
```

Comment:
What is the intended behavior for an operator deregistering when serving in an epoch?  Are they still expected to uphold their commitment to the AVS until the epoch ends?